compass/milestone/MilestoneDatabase.java

```diff
@@ -75,18 +76,46 @@ public MilestoneDatabase(SpongeFactory.Mode powMode, SignatureSource signatureSo
     this.powMode = powMode;
   }
 
+  private static List<String> readInBatches(BufferedReader br, int batchSize, int totalSize) throws IOException {
+    List<String> result = new ArrayList<>(totalSize);
+    for (int i = 1; i < totalSize + batchSize; i += batchSize) {
```

Comment:
Shouldn't the break condition be "read up to totalSize lines" ? See comment below.